src/predict_request_validation_utils.cpp

```diff
@@ -74,7 +73,71 @@ class RequestValidator {
+    const tensor_map_t& inputsInfo;
+    const std::string& servableName;
+    const model_version_t servableVersion;
+    const std::set<std::string>& optionalAllowedInputNames;
+    const Mode batchingMode;
+    const shapes_info_map_t& shapeInfo;
+
+    google::protobuf::internal::RepeatedPtrIterator<const ::inference::ModelInferRequest_InferInputTensor> it;
+
+    RequestValidatorKFS() = delete;
+
+    const std::string& getCurrentlyValidatedInputName() const {
+        return it->name();
+    }
+
+public:
+    RequestValidatorKFS(
+        const ::inference::ModelInferRequest& request, const tensor_map_t& inputsInfo,
+        const std::string& servableName, const model_version_t servableVersion, const std::set<std::string>& optionalAllowedInputNames,
+        const Mode batchingMode, const shapes_info_map_t& shapeInfo) :
+        request(request),
+        inputsInfo(inputsInfo),
+        servableName(servableName),
+        servableVersion(servableVersion),
+        optionalAllowedInputNames(optionalAllowedInputNames),
+        batchingMode(batchingMode),
+        shapeInfo(shapeInfo) {}
+
+    Status validateNumberOfInputs() const;
+    Status validateAndGetInput(const ::inference::ModelInferRequest& request, const std::string& name, google::protobuf::internal::RepeatedPtrIterator<const ::inference::ModelInferRequest_InferInputTensor>& it, size_t& bufferId);
+    Status checkIfShapeValuesNegative(const inference::ModelInferRequest_InferInputTensor& proto) const;
+    Status validateNumberOfBinaryInputShapeDimensions(const inference::ModelInferRequest_InferInputTensor& proto) const;
+    Status checkBatchSizeMismatch(const inference::ModelInferRequest_InferInputTensor& proto, const Dimension& servableBatchSize, const size_t batchSizeIndex, Status& finalStatus, Mode batchingMode, Mode shapeMode) const;
+    Status checkBinaryBatchSizeMismatch(const inference::ModelInferRequest_InferInputTensor& proto, const Dimension& servableBatchSize, Status& finalStatus, Mode batchingMode, Mode shapeMode) const;
+    Status checkShapeMismatch(const inference::ModelInferRequest_InferInputTensor& proto, const ovms::TensorInfo& inputInfo, const size_t batchSizeIndex, Status& finalStatus, Mode batchingMode, Mode shapeMode) const;
+    Status validateTensorContentSize(const inference::ModelInferRequest_InferInputTensor& proto, ovms::Precision expectedPrecision, size_t bufferId) const;
+    Status validateNumberOfShapeDimensions(const ovms::TensorInfo& inputInfo, const inference::ModelInferRequest_InferInputTensor& proto) const;
+    Status validatePrecision(const ovms::TensorInfo& inputInfo, const inference::ModelInferRequest_InferInputTensor& proto) const;
+    Status validate();
+};
+
+Status RequestValidatorKFS::validateNumberOfInputs() const {
+    size_t expectedNumberOfInputs = inputsInfo.size();
+    auto it = request.inputs().begin();
+
+    if (optionalAllowedInputNames.size() > 0) {
+        while (it != request.inputs().end()) {
```

Comment:
Is this piece of code validated in unit tests? If so, the test should not pass since this will be infinite while loop

Reply:
This might indicate we need full scope of unit test for KFS too

Reply:
Good catch - this can be validated only with stateful model support which is planned later.